### ISSUE TITLE
performance issue when using v1.0.10 R Language Support plugin v1.0.10  in Webstorm 2018.3.5 

### ISSUE THREAD

**Issue by UlrichRe** [holgerbrandl/r4intellij#190]
Using the R Language Support plugin in Webstorm 2018.3.5 leads to heavy cpu usage.
Disabling Unused Parameter Inspection and Unused Variable Inspection did not help.

**Comment by holgerbrandl** [comment]
Typically this is just an inital warmup phase during which your installed packaged are indexed.
Am 3/22/2019 9:47:38 PM schrieb UlrichRe <notifications@github.com>:
Using the R Language Support plugin in Webstorm 2018.3.5 leads to heavy cpu usage.
Disabling Unused Parameter Inspection and Unused Variable Inspection did not help.

**Comment by UlrichRe** [comment]
Thank you Holger for your quick reply!

 

I am using your R language Support plugin for quite a while now, and I really like it for the editor features within WebStorm which R Studio does not offer.

 
But there is heavy cpu usage f.i. when editing the same files with WebStorm and RStudio in parallel. The resulting fan noise is disturbing. And my MacBook Pro 2,9 GHz Intel Core i7 is really quiet during all other activities. Would be great to keep it that quiet also when working with your plugin. 
Regards, Ulrich    Am Freitag, 22. März 2019, 21:48:47 MEZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:  
 
 Typically this is just an inital warmup phase during which your installed packaged are indexed.
Am 3/22/2019 9:47:38 PM schrieb UlrichRe <notifications@github.com>:
Using the R Language Support plugin in Webstorm 2018.3.5 leads to heavy cpu usage.
Disabling Unused Parameter Inspection and Unused Variable Inspection did not help.

**Comment by holgerbrandl** [comment]
Does it report ongoing indexing operations (progress bar in the lower right status bar) while you observe high cpu load? 
Am 3/22/2019 9:57:57 PM schrieb UlrichRe <notifications@github.com>:

Thank you Holger for your quick reply!

 

I am using your R language Support plugin for quite a while now, and I really like it for the editor features within WebStorm which R Studio does not offer.

 
But there is heavy cpu usage f.i. when editing the same files with WebStorm and RStudio in parallel. The resulting fan noise is disturbing. And my MacBook Pro 2,9 GHz Intel Core i7 is really quiet during all other activities. Would be great to keep it that quiet also when working with your plugin. 
Regards, Ulrich Am Freitag, 22. März 2019, 21:48:47 MEZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:

Typically this is just an inital warmup phase during which your installed packaged are indexed.
Am 3/22/2019 9:47:38 PM schrieb UlrichRe <notifications@github.com>:
Using the R Language Support plugin in Webstorm 2018.3.5 leads to heavy cpu usage.
Disabling Unused Parameter Inspection and Unused Variable Inspection did not help.

**Comment by UlrichRe** [comment]
Good morning, Holger,
it reports indexing only for a few seconds, but cpu load is still high thereafter (see below).It might have to do with keeping two editor window open in parallel (window -> editor tabs -> split ...)
Regards,Uli
    Am Freitag, 22. März 2019, 22:05:24 MEZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:  
 
 Does it report ongoing indexing operations (progress bar in the lower right status bar) while you observe high cpu load? 
Am 3/22/2019 9:57:57 PM schrieb UlrichRe <notifications@github.com>:

Thank you Holger for your quick reply!

 

I am using your R language Support plugin for quite a while now, and I really like it for the editor features within WebStorm which R Studio does not offer.

 
But there is heavy cpu usage f.i. when editing the same files with WebStorm and RStudio in parallel. The resulting fan noise is disturbing. And my MacBook Pro 2,9 GHz Intel Core i7 is really quiet during all other activities. Would be great to keep it that quiet also when working with your plugin. 
Regards, Ulrich Am Freitag, 22. März 2019, 21:48:47 MEZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:

Typically this is just an inital warmup phase during which your installed packaged are indexed.
Am 3/22/2019 9:47:38 PM schrieb UlrichRe <notifications@github.com>:
Using the R Language Support plugin in Webstorm 2018.3.5 leads to heavy cpu usage.
Disabling Unused Parameter Inspection and Unused Variable Inspection did not help.

**Comment by holgerbrandl** [comment]
Thanks for the feedback. I'll keep an eye on it, but for now I'm afraid there's no direct path for me to reproduce and fix the issue.

**Comment by UlrichRe** [comment]
yes, thank you! That would be fine if the issue could be solved.
Regards,Uli
    Am Montag, 25. März 2019, 15:12:01 MEZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:  
 
 
Thanks for the feedback. I'll keep an eye on it, but for now I'm afraid there's no direct path for me to reproduce and fix the issue.

**Comment by UlrichRe** [comment]
Holger,
maybe the high cpu load is due to permanent error analysis?In my application f.i. WebStorm reports 288 errors, 808 warnings and 1001 typos (see attached image) although the R code works fine.
Regards,Ulrich




    Am Montag, 25. März 2019, 15:12:01 MEZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:  
 
 
Thanks for the feedback. I'll keep an eye on it, but for now I'm afraid there's no direct path for me to reproduce and fix the issue.

**Comment by holgerbrandl** [comment]
Indeed that's likely. I've also noticed for other language plugins: With many errors, IJ becomes very slow

**Comment by UlrichRe** [comment]
WebStorm reports many cases of 'Forward Reference' and 'Unsolved Reference' although, as I wrote, the R code works fine.
Is there any way to exclude error analysis when using your plugin?
I would mainly like to use WebStorm for R as an extended editor with better search/replace functionality than RStudio.
IJ?
Regards,Ulrich 
    Am Sonntag, 20. Oktober 2019, 20:43:30 MESZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:  
 
 
Indeed that's likely. I've also noticed for other language plugins: With many errors, IJ becomes very slow

**Comment by holgerbrandl** [comment]
Sure, you can always disable resolver errors. General parsing errors can not be disabled, but in case you spot some of these, you're welcome to file a more specific issue.

**Comment by UlrichRe** [comment]
Thanks for the hint.I managed to get rid of almost all warnings and error messages by disabling all flags in WebStorm -> Preferences -> Editor -> Inspections -> R.
The remaining three errors referred to using ** instead of ^. (** is actually allowed.)
Regards,Ulrich

    Am Dienstag, 22. Oktober 2019, 12:10:26 MESZ hat Holger Brandl <notifications@github.com> Folgendes geschrieben:  
 
 
Sure, you can always disable resolver errors. General parsing errors can not be disabled, but in case you spot some of these, you're welcome to file a more specific issue.
